modules/Microsoft.RecoveryServices/vaults/deploy.bicep

```diff
@@ -173,6 +180,7 @@ resource rsv 'Microsoft.RecoveryServices/vaults@2022-09-10' = {
   properties: {
     monitoringSettings: !empty(monitoringSettings) ? monitoringSettings : null
     securitySettings: !empty(securitySettings) ? securitySettings : null
+    publicNetworkAccess: !empty(publicNetworkAccess) ? publicNetworkAccess : null
```

Comment:
Hey @juanandmsft thanks for this addition, very useful functionality indeed! Please make sure to align with @ChrisSidebotham as he previously opened a PR which is related #2488